mirrord/agent/src/main.rs

```diff
@@ -446,7 +444,11 @@ async fn start_agent() -> Result<()> {
     .await?;
 
     let mut state = State::new(&args).await?;
-    let pid = state.get_container_pid().await?;
+    let (pid, env) = match state.get_container_info().await? {
```

Comment:
This`?` now means the agent exits if it can't get the environment variables, right? Before it would return a remote error to the client, but wouldn't exit directly because of it. Just making sure this change is intentional.

Reply:
yeah but if the error occurred the layer would die anyway so don't think it's a massive change. also there's no reason for it to fail